osprey/processes/wps_convolution.py

```diff
@@ -72,38 +123,69 @@ def __init__(self):
             status_supported=True,
         )
 
+    def config_dict_hander(self, config):
+        """
+        This function enables users to provide dictionary-like string for Configuration input.
+        If CASE_DIR and REST_DATE are not provided from a user, their values are derived from
+        CASEID and STOP_DATE by default.
+        """
+        input_dict = json.loads(config)
+        try:
+            for upper_key in input_dict.keys():
+                for lower_key in input_dict[upper_key].keys():
+                    self.config_dict[upper_key][lower_key] = input_dict[upper_key][
+                        lower_key
+                    ]
+
+            if self.config_dict["OPTIONS"]["CASE_DIR"] == None:
+                self.config_dict["OPTIONS"]["CASE_DIR"] = os.path.join(
+                    self.workdir, self.config_dict["OPTIONS"]["CASEID"]
+                )
+            if self.config_dict["OPTIONS"]["REST_DATE"] == None:
+                self.config_dict["OPTIONS"]["REST_DATE"] = self.config_dict["OPTIONS"][
+                    "STOP_DATE"
+                ]
+
+        except KeyError as e:
+            raise ProcessError(f"Invalid config key provided")
+
+    def config_file_builder(self, config_dict):
+        """
+        This function is used for RVIC1.1.0.post1 only since the version requires Configuration input
+        to be a .cfg filepath. The function uses information from config_dict to create the file.
+        """
+        cfg_filepath = os.path.join(self.workdir, "convolve_file.cfg")
```

Comment:
This config name should be a configurable, no? Unless the only file `rvic` looks at is one with this name.

Reply:
rvic only requires a .cfg file for the input. No other files are required, so I believe it is ok to leave it like that.